flocker/control/_protocol.py

```diff
@@ -109,6 +109,11 @@ class NodeStateCommand(Command):
     Used by a convergence agent to update the control service about the
     status of a particular node.
     """
+    # FLOC-1513
+    #
+    # Add another argument, an optional `ListOf(Unicode)` representing
```

Comment:
We serialize any object as long as it's on the approved list ... and an instance of the class passed to `SerializableArgument`. And it's not arbitrary objects, it's "IClusterStateUpdater instances that are on the approved list of serializable objects".

So basically you have a variant of `SerializableArgument` which takes a list of objects and checks their interface, and then relies on the approved serializable list after that.